### ISSUE TITLE
Valid module arguments are reported as `args[module]` violations in ansible-lint 6.10.0

### ISSUE THREAD

**Issue by piotrp** [ansible/ansible-lint#2830]
ansible-lint 6.10.0 doesn't recognize `use_ssh_args` argument of `synchronize` module (but recognizes nonexistent `ssh_args`).

Playbook:
```yml
- name: Test
  hosts: localhost
  tasks:
    - name: Copy files
      synchronize:
        src: .
        dest: .
        owner: false
        group: false
        use_ssh_args: true
```

Output:
```
$ ansible-lint synchronize-test.yml
WARNING  Skipped installing collection dependencies due to running in offline mode.
WARNING  Listing 1 violation(s) that are fatal
args[module]: Unsupported parameters for synchronize module: use_ssh_args. Supported parameters include: set_remote_user, rsync_opts, times, copy_links, _local_rsync_path, recursive, rsync_path, rsync_timeout, dirs, delete, group, src, compress, dest_port, ssh_connection_multiplexing, delay_updates, ssh_args, partial, link_dest, _local_rsync_password, _substitute_controller, verify_host, links, owner, mode, existing_only, checksum, archive, perms, private_key, dest. (warning)
synchronize-test.yml:4 Task/Handler: Copy files

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

                  Rule Violation Summary
 count tag          profile rule associated tags
     1 args[module]         syntax, experimental (warning)

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files.
```

I also see an issue in how things are worded in output: ansible-lint prints "Listing 1 violation(s) that are fatal" but also "Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files." and exits with `0` - so it it fatal or not?

**Comment by nre-ableton** [comment]
This is almost certainly caused by https://github.com/ansible/ansible-lint/pull/2749. I'm experiencing the same thing with the `service` module and the `use` parameter (which, similar to your example, is supported).

I don't really have time to dig into the code to figure out what is going wrong, but I suspect that the problem happens around [this area of the code](https://github.com/ansible/ansible-lint/blob/40155c89eadb5de2bb47cc6e14d76e7348578d75/src/ansiblelint/rules/args.py#L89) where `args.py` attempts to load the module from the spec.

@piotrp Would you mind making your issue title a bit more generic? I have a feeling that this problem affects a _lot_ more modules than just the two that we have noticed so far.

**Comment by nre-ableton** [comment]
@piotrp Thanks! :+1:

**Comment by evgeni** [comment]
I think this is another case where you're calling the `synchronize` *action* but `ansible-lint` checks it against the `synchronize` *module*, which has different semantics. I've reported that in #2824

**Comment by ssbarnea** [comment]
@ganeshrn I can confirm that bug, ansible-lint displays failure to validate module args with `WARNING  Unable to load module ansible.posix.synchronize at ... for options validation` while the arguments are valid!

**Comment by ganeshrn** [comment]
@piotrp @evgeni Thanks for reporting the issue.
@nre-ableton You are spot on with your analysis. Without a public API for core, it becomes tricky to validate module arg spec for static analysis, especially in cases with special handling for arguments done in the action plugin or mismatch in what is there is the module documentation and what is passed to the `AnsibleModule` class.
We will try to figure out a workaround for such scenarios.

@ssbarnea Do you have suggestions on how it can be done?

**Comment by ssbarnea** [comment]
Unable to reproduce with current code from main branch, probably fixed recently.

**Comment by piotrp** [comment]
When will the fix be released? It's not in 6.14.0 that was released two days ago.
```
$ ansible-lint --version
ansible-lint 6.14.0 using ansible 2.14.2
```
```
$ ansible-galaxy collection list

# /e/ansible-lint-args-test/.venv/ansible_collections
Collection    Version
------------- -------
ansible.posix 1.5.1
```
```
$ cat synchronize-test.yml
- name: Test
  hosts: localhost
  tasks:
    - name: Copy files
      ansible.posix.synchronize:
        src: .
        dest: .
        owner: false
        group: false
        use_ssh_args: true
```
```
$ ansible-lint synchronize-test.yml
WARNING  Listing 1 violation(s) that are fatal
args[module]: Unsupported parameters for (basic.py) module: use_ssh_args. Supported parameters include: _local_rsync_password, _local_rsync_path, _substitute_controller, archive, checksum, compress, copy_links, delay_updates, delete, dest, dest_port, dirs, existing_only, group, link_dest, links, mode, owner, partial, perms, private_key, recursive, rsync_opts, rsync_path, rsync_timeout, set_remote_user, src, ssh_args, ssh_connection_multiplexing, times, verify_host. (warning)
synchronize-test.yml:4 Task/Handler: Copy files

Read documentation for instructions on how to ignore specific rule violations.

                  Rule Violation Summary
 count tag          profile rule associated tags
     1 args[module]         syntax, experimental (warning)

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 1 files.
```

**Comment by nre-ableton** [comment]
Actually, it seems that this was fixed in 6.13.1 but broken again in 6.14.0. I'll open a new issue for the regression, since I'm not sure whether the maintainers are paying attention to closed issues.

**Comment by piotrp** [comment]
Reposted as #3152, this issue is still present in 6.14.1, i.e. in latest commit on the main branch.
